include/tvm/VariableVector.h

```diff
@@ -75,20 +83,29 @@ class TVM_DLLAPI VariableVector
   void add(const std::vector<VariablePtr> & variables);
   /** Same as add(VariablePtr), but for adding a vector of variables.*/
   void add(const VariableVector & variables);
-  /** Add a variable to the vector, if not already present and return the index
+  /** Add a variable to the vector, if not already present, and return the index
    * of the variable in the vector whether it was added or not.
    *
    * \param v The variable to be added.
+   * \param containingIndex Specify what to return in case \p v is the subvariable
+   * of an already present variable. If \c false, return -1 in this case. If
+   * \c true, return the index of the variable containing \p v.
    *
-   * \returns Index of variable \p v in the vector.
+   * \returns Index of variable \p v in the vector. If \p v is a subvariable of an
+   * already present variable, returns -1 or the index of the variable depending
+   * on \p containingIndex.
    */
-  int addAndGetIndex(VariablePtr v);
+  int addAndGetIndex(VariablePtr v, bool containingIndex = false);
```

Comment:
The use I was doing of this function, introduced for the update of the problems, has been indeed replaced by the range counting utilities.
Yet I think it can still be useful, and, given that it is a bit tricky and tested, I would be in favor of keeping it.